src/commands/lambda/flare.ts

```diff
@@ -1,14 +1,229 @@
+import * as fs from 'fs'
+import * as path from 'path'
+import util from 'util'
+
+import {LambdaClient, LambdaClientConfig} from '@aws-sdk/client-lambda'
+import {AwsCredentialIdentity} from '@aws-sdk/types'
+import axios from 'axios'
 import {Command} from 'clipanion'
+import FormData from 'form-data'
+import JSZip from 'jszip'
 
+import {API_KEY_ENV_VAR, AWS_DEFAULT_REGION_ENV_VAR, CI_API_KEY_ENV_VAR} from './constants'
+import {getAWSCredentials, getLambdaFunctionConfig, getRegion} from './functions/commons'
+import {requestAWSCredentials} from './prompt'
+import {
+  renderError,
+  renderNoAWSCredentialsFound,
+  renderNoDefaultRegionSpecifiedError,
+  renderSoftWarning,
+} from './renderers/common-renderer'
 import {renderLambdaFlareHeader} from './renderers/flare-renderer'
 
 export class LambdaFlareCommand extends Command {
   private isDryRun = false
+  private functionName?: string
+  private region?: string
+  private apiKey?: string
+  private caseId?: string
+  private email?: string
+  private credentials?: AwsCredentialIdentity
 
+  /**
+   * Entry point for the `lambda flare` command.
+   * Gathers lambda function configuration and sends it to Datadog.
+   * @returns 0 if the command ran successfully, 1 otherwise.
+   */
   public async execute() {
     this.context.stdout.write(renderLambdaFlareHeader(this.isDryRun))
+
+    // Validate function name
+    let errorFound = false
+    if (this.functionName === undefined) {
+      this.context.stderr.write(renderError('No function name specified. [-f,--function]'))
+      errorFound = true
+    }
+
+    // Validate region
+    this.region = this.region ?? getRegion(this.functionName ?? '')
+    this.region = this.region ?? process.env[AWS_DEFAULT_REGION_ENV_VAR]
+    if (this.region === undefined) {
+      this.context.stderr.write(renderNoDefaultRegionSpecifiedError())
+      errorFound = true
+    }
```

Comment:
Let's refactor this code based on our conversation – let's make this a constant, keeping the provided customer value in `this.region`. 
Also, let's prioritize this way: 
1. Region from ARN.
2. Region from `this.region`.
3. AWS default region.